### DISCUSSION TITLE
Listbox `onBlur` prop

### DISCUSSION THREAD

**Discussion by dreamawakening** [tailwindlabs/headlessui#796] [Ideas]
Some libraries like Formik use this in fields. There should be a prop for this.

**Comment by Shahaed** [comment]
I agree. All props exposed by native html select should be exposed for listbox

**Reply by dreamawakening** [reply]
@Shahaed did you figure out a workaround?

**Reply by Shahaed** [reply]
Yeah, I hooked it up to formik fairly easily. Here is a gist I wrote up about how: https://gist.github.com/Shahaed/6094a3a971fc3afd7f82c8b1ddcc71cc

Using the `useField` hook you should be able to hook up every formik functionality. https://formik.org/docs/api/useField

@dspacejs

**Reply by dreamawakening** [reply]
@Shahaed perfect thanks, I didn't realise `Listbox.Button` could take an `onBlur` prop.

**Reply by Shahaed** [reply]
If you check the headless docs each of the elements renders as an html element. 

> For example, Listbox.Label renders a label by default, Listbox.Button renders a button, Listbox.Options renders a ul, and Listbox.Option renders a li. Listbox interestingly does not render an extra element, and instead renders its children directly.

So they all have some native html events you can plug into.

If you find a better way than mine, let me know @dspacejs
